OpenRA.Mods.Common/Activities/DeliverUnit.cs

```diff
@@ -18,44 +18,34 @@ namespace OpenRA.Mods.Common.Activities
 	public class DeliverUnit : Activity
 	{
 		readonly Actor self;
-		readonly Carryable carryable;
 		readonly Carryall carryall;
-		readonly IPositionable positionable;
 		readonly BodyOrientation body;
-		readonly IFacing carryableFacing;
 		readonly IFacing carryallFacing;
-		readonly CPos destination;
+		CPos? destination;
 
-		enum DeliveryState { Transport, Land, Wait, Release, TakeOff, Done, Aborted }
-
-		DeliveryState state;
-
-		public DeliverUnit(Actor self, CPos destination)
+		public DeliverUnit(Actor self, CPos? destination = null)
 		{
 			this.self = self;
 			this.destination = destination;
 
 			carryallFacing = self.Trait<IFacing>();
 			carryall = self.Trait<Carryall>();
 			body = self.Trait<BodyOrientation>();
-
-			carryable = carryall.Carryable.Trait<Carryable>();
-			positionable = carryall.Carryable.Trait<IPositionable>();
-			carryableFacing = carryall.Carryable.Trait<IFacing>();
-			state = DeliveryState.Transport;
 		}
 
 		CPos? FindDropLocation(CPos targetCell, WDist maxSearchDistance)
```

Comment:
We can make the `destination` and `targetCell` handling tidier by changing the `CPos?` to `Target` - see f114f27b438f960ab1643d89142ed524e9d22543

Reply:
Cherry-picked.